src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsMSBuildProjectSystem.cs

```diff
@@ -139,10 +139,10 @@ private async Task AddFileCoreAsync(string path, Action addFile)
             var lockFileFullPath = PackagesConfigLockFileUtility.GetPackagesLockFilePath(ProjectFullPath, GetPropertyValue("NuGetLockFilePath")?.ToString(), ProjectName);
             if (File.Exists(Path.Combine(ProjectFullPath, path))
                 && !fileExistsInProject
-                && !fileName.Equals(ProjectManagement.Constants.PackageReferenceFile)
-                && !fileName.Equals("packages." + ProjectName + ".config")
-                && !fileName.Equals(EnvDteProjectExtensions.WebConfig)
-                && !fileName.Equals(EnvDteProjectExtensions.AppConfig)
+                && !fileName.Equals(ProjectManagement.Constants.PackageReferenceFile, StringComparison.OrdinalIgnoreCase)
```

Comment:
Path comparisons should be aware of the FS casing. 

This applies through the whole PR.

Reply:
Here the comparison in for the file name only, should this also be aware of the OS?